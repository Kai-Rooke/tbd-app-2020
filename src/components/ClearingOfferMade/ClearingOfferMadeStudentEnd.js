import React from "react";
import TextBox from "../../components/TextBox/TextBox";
import "./ClearingOfferMade.css";
import { Container, Col, Row } from "react-bootstrap";
import UCPButton from "../Buttons/UCPButton";


const ClearingOfferMadeFifthPage = (props) => {
   function AutoInputData(){
    if(global.ApplicationFormData.email !== "")
    {
      var email = document.getElementById("emailConfirm");
      email.value = global.ApplicationFormData.email;
    }
    if( global.ApplicationFormData.selectedCourse !== "")
    {
        var course = document.getElementById("courseApplied");
        course.value = global.ApplicationFormData.email;
    }
   }
   function DownloadFile()
   {
        var saveData = (function () { 
        var a = document.createElement("a"); 
            document.body.appendChild(a); 
            a.style = "display: none"; 
    return function (fileName) { 
            var json = JSON.stringify(global.ApplicationFormData), 
            blob = new Blob([json], {type: "octet/stream"}), 
            url = window.URL.createObjectURL(blob); 
        a.href = url; 
        a.download = fileName; 
        a.click(); 
        window.URL.revokeObjectURL(url); 
    }; 
    }()); 
    saveData(fileName)
   }
 
var fileName = "studentDetails.json"; 
 
 


return ( <Container>
    <h1>Thank you for using the form</h1>
    <p>Please press confirm to complete</p>
    <h3>Please confirm email that your application will be sent to</h3>
    <TextBox id="emailConfirm" placeholder={global.ApplicationFormData.email}/>
   
<Row>
    <Col><br></br><UCPButton to='/ClearingOfferMadeFourthPage'
                primary="True"
                className="mediumbutton"
                buttonText="Go Back"
                href="/ClearingOfferMadeFourthPage"></UCPButton></Col>

    <Col><br></br>
    <div onClick={DownloadFile}>
        <UCPButton to ='/'
                 primary="True"
                className="mediumbutton"
                buttonText="Confirm"
                href="/"></UCPButton>
    </div>
        </Col>
    </Row>
    {AutoInputData}
 </Container>
)
}
export default ClearingOfferMadeFifthPage;