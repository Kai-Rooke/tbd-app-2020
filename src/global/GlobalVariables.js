global.ApplicationFormData = [];
global.userType = "student";
global.staffFirstName = "";
global.staffLastName = "";

