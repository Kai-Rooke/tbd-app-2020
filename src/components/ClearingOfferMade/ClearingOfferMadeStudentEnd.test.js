import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import ClearingOfferMadeStudentEnd from './ClearingOfferMadeStudentEnd';
import { BrowserRouter as Router} from 'react-router-dom';
import '../../global/GlobalVariables.js';

//cleans up 
afterEach(cleanup);

it("Renders ClearingOfferMadeStudentEnd", () => {
    global.ApplicationFormData.email = "Test@email.com";
    const { asFragment } = render(<Router><ClearingOfferMadeStudentEnd /></Router>);
    expect(asFragment()).toMatchSnapshot();
});