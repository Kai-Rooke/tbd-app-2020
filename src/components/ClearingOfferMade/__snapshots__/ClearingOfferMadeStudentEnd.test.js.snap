// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ClearingOfferMadeStudentEnd 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      Thank you for using the form
    </h1>
    <p>
      Please press confirm to complete
    </p>
    <h3>
      Please confirm email that your application will be sent to
    </h3>
    <div
      class="mobiletextbox"
    >
      <label
        for="emailConfirm"
      />
      <input
        id="emailConfirm"
        maxlength="30"
        minlength="0"
        placeholder="Test@email.com"
        size="20"
        type="text"
        value=""
      />
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <br />
        <a
          href="/DegreeQualifications"
        >
          <button
            class="mediumbutton"
            data-testid="ucpButton"
            href="/DegreeQualifications"
            primary="True"
            type="button"
          >
            Go Back​
          </button>
        </a>
      </div>
      <div
        class="col"
      >
        <br />
        <div>
          <a
            href="/"
          >
            <button
              class="mediumbutton"
              data-testid="ucpButton"
              href="/"
              primary="True"
              type="button"
            >
              Confirm​
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
